test/unit/modules.spec.js

```diff
@@ -124,6 +125,49 @@ describe('Modules', () => {
       store.commit('a/foo')
       expect(mutationSpy).toHaveBeenCalled()
     })
+
+    it('should keep getters when component gets destroyed', async () => {
+      const store = new Vuex.Store()
+
+      const moduleA = {
+        namespaced: true,
+        state: () => ({ value: 1 }),
+        getters: {
+          getState: (state) => state.value
+        },
+        mutations: {
+          increment: (state) => { state.value++ }
+        }
+      }
+
+      const CompA = {
+        template: `<div />`,
+        created () {
+          this.$store.registerModule('moduleA', moduleA)
+        }
+      }
+
+      const CompB = {
+        template: `<div />`
+      }
+
+      const vm = mount(store, {
+        components: { CompA, CompB },
+        data: () => ({ show: 'a' }),
+        render () {
+          return this.show === 'a' ? h(CompA) : h(CompB)
+        }
+      })
+
+      expect(store.getters['moduleA/getState']).toBe(1)
+
+      vm.show = 'b'
+      await nextTick()
+
+      store.commit('moduleA/increment')
+
+      expect(store.getters['moduleA/getState']).toBe(2)
```

Comment:
You can also destroy the `vm` and mount it again

Reply:
Yeah I tried it but for some reason I couldn't get destroy to work 🤔  I don't remember why it was long time ago but 😳